src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx

```diff
@@ -178,43 +178,63 @@ export class SavedObjectsTable extends Component<SavedObjectsTableProps, SavedOb
   }
 
   fetchCounts = async () => {
-    const { allowedTypes } = this.props;
+    const { allowedTypes, namespaceRegistry } = this.props;
     const { queryText, visibleTypes, visibleNamespaces } = parseQuery(this.state.activeQuery);
 
     const filteredTypes = allowedTypes.filter(
       (type) => !visibleTypes || visibleTypes.includes(type)
     );
 
+    let availableNamespaces = namespaceRegistry.getAll() || [];
+    availableNamespaces = availableNamespaces.map((ns) => ns.id);
+
+    const filteredCountOptions = {
+      typesToInclude: filteredTypes,
+      searchString: queryText,
+    };
+
+    if (availableNamespaces?.length) {
+      if (visibleNamespaces?.length) {
+        filteredCountOptions.namespacesToInclude = visibleNamespaces;
```

Comment:
If `visibleNamespaces` is used after all, do we need to care for `availableNamespaces?.length`?

Reply:
The way I was thinking about this is:

```
if (registeredNamespaces) {
  if (namespacesSelected): query by selected namespaces
  else: query by all registered namespaces
}
```

If visibleNamespaces ever has elements then it should already imply that namespaces are registered.